File: susupay-frontend/src/types/transaction.ts
export type SubmissionType = 'SMS_TEXT' | 'SCREENSHOT';
export type TrustLevel = 'HIGH' | 'MEDIUM' | 'LOW' | 'AUTO_REJECTED';
export type TransactionStatus = 'PENDING' | 'CONFIRMED' | 'QUERIED' | 'REJECTED' | 'AUTO_REJECTED';
export type Confidence = 'HIGH' | 'PARTIAL' | 'FAILED';

export interface SMSSubmitRequest {
  client_id: string;
  sms_text: string;
}

export interface ClientSMSSubmitRequest {
  sms_text: string;
}

export interface ScreenshotSubmitRequest {
  client_id: string;
  amount: number;
  screenshot: File;
}

export interface ParsedSMSResponse {
  amount: number | null;
  recipient_name: string | null;
  recipient_phone: string | null;
  transaction_id: string | null;
  transaction_date: string | null;
  confidence: Confidence;
}

export interface ValidationFlag {
  check: string;
  passed: boolean;
  detail: string;
}

export interface SubmitResponse {
  transaction_id: string;
  status: TransactionStatus;
  trust_level: TrustLevel;
  validation_flags: ValidationFlag[];
  parsed: ParsedSMSResponse;
}

export interface TransactionFeedItem {
  id: string;
  client_id: string;
  client_name: string;
  amount: string;
  submission_type: SubmissionType;
  trust_level: TrustLevel;
  status: TransactionStatus;
  validation_flags: ValidationFlag[];
  submitted_at: string;
  confirmed_at: string | null;
  collector_note: string | null;
}

export interface QueryRequest {
  note: string;
}

export interface RejectRequest {
  note: string;
}

export interface TransactionActionResponse {
  transaction_id: string;
  status: TransactionStatus;
  confirmed_at: string | null;
}

export interface ClientTransactionItem {
  id: string;
  amount: string;
  status: TransactionStatus;
  trust_level: TrustLevel;
  submitted_at: string;
  confirmed_at: string | null;
  collector_note: string | null;
}

File: susupay-backend/app/schemas/transaction.py
import uuid
from datetime import datetime
from decimal import Decimal

from pydantic import BaseModel, Field


# --- Submission ---


class SMSSubmitRequest(BaseModel):
    client_id: uuid.UUID
    sms_text: str = Field(..., min_length=10, max_length=2000)


class ClientSMSSubmitRequest(BaseModel):
    """Client submits their own SMS — no client_id needed (from JWT)."""
    sms_text: str = Field(..., min_length=10, max_length=2000)


class ScreenshotSubmitRequest(BaseModel):
    client_id: uuid.UUID
    amount: float = Field(..., gt=0)
    # screenshot_key will be set after S3 upload in Phase 4


class ParsedSMSResponse(BaseModel):
    amount: float | None
    recipient_name: str | None
    recipient_phone: str | None
    transaction_id: str | None
    transaction_date: datetime | None
    confidence: str


class SubmitResponse(BaseModel):
    transaction_id: uuid.UUID
    status: str
    trust_level: str
    validation_flags: list[dict] | None = None
    parsed: ParsedSMSResponse | None = None


# --- Feed ---


class TransactionFeedItem(BaseModel):
    id: uuid.UUID
    client_id: uuid.UUID
    client_name: str
    amount: Decimal
    submission_type: str
    trust_level: str
    status: str
    validation_flags: list[dict] | None = None
    submitted_at: datetime
    confirmed_at: datetime | None = None
    collector_note: str | None = None

    model_config = {"from_attributes": True}


# --- Actions ---


class ConfirmRequest(BaseModel):
    pass


class QueryRequest(BaseModel):
    note: str = Field(..., min_length=1, max_length=500)


class RejectRequest(BaseModel):
    note: str = Field(..., min_length=1, max_length=500)


class TransactionActionResponse(BaseModel):
    transaction_id: uuid.UUID
    status: str
    confirmed_at: datetime | None = None


# --- Client history ---


class ClientTransactionItem(BaseModel):
    id: uuid.UUID
    amount: Decimal
    status: str
    trust_level: str
    submitted_at: datetime
    confirmed_at: datetime | None = None
    collector_note: str | None = None

    model_config = {"from_attributes": True}
